server/plugin/api.go

```diff
@@ -317,31 +331,72 @@ func (p *MatterpollPlugin) publishPollMetadata(poll *poll.Poll, userID string) {
 		p.API.LogWarn("Failed to check admin permission", "userID", userID, "pollID", poll.ID, "error", appErr.Error())
 		hasAdminPermission = false
 	}
-	metadata, err := poll.GetMetadata(userID, hasAdminPermission)
+	metadata := poll.GetMetadata(userID, hasAdminPermission)
+	p.API.PublishWebSocketEvent("has_voted", metadata.ToMap(), &model.WebsocketBroadcast{UserId: userID})
+}
+
+func (p *MatterpollPlugin) handleResetVotes(vars map[string]string, request *model.PostActionIntegrationRequest) (*i18n.LocalizeConfig, *model.Post, error) {
+	pollID := vars["id"]
+	userID := request.UserId
+
+	poll, err := p.Store.Poll().Get(pollID)
 	if err != nil {
-		p.API.LogWarn("Failed to get poll metadata", "userID", userID, "pollID", poll.ID, "error", appErr.Error())
-		return
+		return &i18n.LocalizeConfig{DefaultMessage: commandErrorGeneric}, nil, errors.Wrap(err, "failed to get poll")
 	}
 
-	p.API.PublishWebSocketEvent("has_voted", metadata.ToMap(), &model.WebsocketBroadcast{UserId: userID})
+	displayName, appErr := p.ConvertCreatorIDToDisplayName(poll.Creator)
+	if appErr != nil {
+		return &i18n.LocalizeConfig{DefaultMessage: commandErrorGeneric}, nil, errors.Wrap(appErr, "failed to get display name for creator")
+	}
+
+	votedAnswers := poll.GetVotedAnswers(userID)
+	if len(votedAnswers) == 0 {
+		return &i18n.LocalizeConfig{DefaultMessage: &i18n.Message{
+			ID:    "response.resetVotes.noVotes",
+			Other: "There are no votes to reset.",
+		}}, nil, nil
+	}
+
+	prev := poll.Copy()
+
+	poll.ResetVotes(userID)
+
+	if err = p.Store.Poll().Update(prev, poll); err != nil {
+		return &i18n.LocalizeConfig{DefaultMessage: commandErrorGeneric}, nil, errors.Wrap(err, "failed to save poll")
+	}
+
+	go p.publishPollMetadata(poll, userID)
+
+	post := &model.Post{}
+	publicLocalizer := p.getServerLocalizer()
+	model.ParseSlackAttachment(post, poll.ToPostActions(publicLocalizer, manifest.ID, displayName))
+	post.AddProp("poll_id", poll.ID)
+
+	return &i18n.LocalizeConfig{
+		DefaultMessage: &i18n.Message{
+			ID:    "response.resetVotes.success",
+			Other: "All votes are cleared. [{{.ClearedVotes}}]",
```

Comment:
I'm wondering what the value of printing the previous answers is. The user already reset them, why do we print them?

Reply:
That isn't necessary, but I imagined that there would be cases where voter would want to see where voter had voted after resetting, such as if voter wanted to change only one of the many voted options. Should remove it?

Reply:
I'm 2/5 that the current text isn't great. If we keep it, can we change it to something like `All votes are cleared. Your previous votes were {{.ClearedVotes}}.`?

Reply:
It looks so good! I've fixed as you suggested. Thanks.